src/mame/heathkit/tlb.cpp

```diff
@@ -0,0 +1,635 @@
+// license:BSD-3-Clause
+// copyright-holders:Mark Garlanger
+/***************************************************************************
+
+		Heathkit Terminal Logic Board (TLB)
+
+		The board used in the H19 smart terminal designed and manufactured
+		by Heath Company. (and identical Z19 sold by Zenith Data Systems)
+
+		The keyboard consists of a 9x10 matrix connected to a MM5740AAC/N
+		mask-programmed keyboard controller. The output of this passes
+		through a rom.
+
+		Input can also come from the serial port (a 8250).
+		Either device will signal an interrupt to the CPU when a key
+		is pressed/data is received.
+
+		TODO:
+		- speed up emulation
+		- update SW401 baud rate options for Watz ROM
+		- update SW401 & SW402 definitions for Super-19 ROM
+		- update SW401 & SW402 definitions for ULTRA ROM
+		- add option for ULTRA ROMs second page of screen RAM
+
+****************************************************************************/
+/***************************************************************************
+ Memory Layout
+	 The U435 three-to-eight line decoder uses A14 and A15 to generate three memory addresses:
+
+	 1.   Program ROM        0x0000
+
+	 2.   Scratchpad RAM     0x4000
+
+	 3.   Display Memory     0xF800
+
+
+ Port Layout
+
+	 Only address lines A5, A6, A7 are used by the U442 three-to-eight line decoder
+
+Address   Description
+----------------------------------------------------
+ 0x00   Power-up configuration (primary - SW401)
+ 0x20   Power-up configuration (secondary - SW402)
+ 0x40   ACE (communications)
+ 0x60   CRT controller
+ 0x80   Keyboard encoder
+ 0xA0   Keyboard status
+ 0xC0   Key click enable
+ 0xE0   Bell enable
+
+****************************************************************************/
+
+#include "tlb.h"
+
+// Standard H19 used a 2.048 MHz clock
+#define H19_CLOCK (XTAL(12'288'000) / 6)
+#define MC6845_CLOCK (XTAL(12'288'000) /8)
+#define INS8250_CLOCK (XTAL(12'288'000) /4)
+
+// Capacitor value in pF
+#define H19_KEY_DEBOUNCE_CAPACITOR 5000
+#define MM5740_CLOCK (mm5740_device::calc_effective_clock_key_debounce(H19_KEY_DEBOUNCE_CAPACITOR))
+
+// Beep Frequency is 1 KHz
+#define H19_BEEP_FRQ (H19_CLOCK / 2048)
+
+DEFINE_DEVICE_TYPE(TLB, heath_tlb_device, "heath_tlb", "Heath Terminal Logic Board");
+DEFINE_DEVICE_TYPE(SUPER19, heath_super19_tlb_device, "heath_super19_tlb", "Heath Terminal Logic Board w/Super19 ROM");
+DEFINE_DEVICE_TYPE(WATZ, heath_watz_tlb_device, "heath_watz_tlb", "Heath Terminal Logic Board w/Watzman ROM");
+DEFINE_DEVICE_TYPE(ULTRA, heath_ultra_tlb_device, "heath_ultra_tlb", "Heath Terminal Logic Board w/Ultra ROM");
```

Comment:
The device type names are far too generic.  Remember they’re in the global namespace.